### PR TITLE
Add control buffers to java build

### PR DESCRIPTION
**Pull request by dsyer**
Also make Java build work without an explicit target directory and
add instructions in README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,4 +60,4 @@ protoc --doc_out=markdown,README.md:. *.proto
 Dropsonde is intended to be a "fire and forget" protocol, in the sense that an emitter should send events to its receiver with no expectation of acknowledgement. There is no "handshake" step; the emitter simply begins emitting to a known address of an expected recipient. 
 
 ### Heartbeat requests
-One of the control messages available is the [HeartbeatRequest](control/README.md#control.HeartbeatRequest). When a process implementing Dropsonde receives this message, it MUST respond with with a [Heartbeat](events/README.md#events.Heartbeat) message. That `Heartbeat` should include the same `UUID` as was received in the containing `ControlMessage`.
```
Comment (jmtuley) [inline README.md:54]: Please remove this change; we'd like to keep our final newlines per UNIX standards.

File: generate-java.sh [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 dir_resolve()
 {
+"$1"!="" && mkdir -p "$1"
```
Comment (jmtuley) [inline generate-java.sh:5]: Any particular reason to make the target optional for the Java generator? Seems to me we'd like parity with the Go generator; both or neither should make the target optional.

File: README.md [hunk 3]
```diff
@@ -29,10 +29,19 @@ Please see the following for detailed descriptions of each type:
    ```
    ./generate-go.sh TARGET_PATH
    ```
+   
+### Java
+
+1. Build the go code first (see above) so that all the imports are available
```
Comment (jmtuley) [inline README.md:35]: Is there a way we could make this step unnecessary? Requiring a Java developer to have Go and all of the associated tooling available seems less than optional.

On the other hand, I know nothing about the Java tooling, so maybe this really is necessary?

File: README.md (hunk 1 above)
Comment (dsyer) [inline README.md:54]: I can't tell what the change is from the github UI, and I don't know what the "standard" is. I'm removing the trailing newline, hoping that was what you meant.

File: generate-java.sh (hunk 2 above)
Comment (dsyer) [inline generate-java.sh:5]: The target is not mandatory for the go generator, so actually I was aligning the two with this change.

File: README.md (hunk 3 above)
Comment (dsyer) [inline README.md:35]: There probably is a way, but if so it seems like it would deserve another pull request. As it stands isn't it better to have something that works (as opposed to what is currently in master)?

Comment (dsyer) [inline README.md:35]: Done (but if that was the "standard" then I have to say it was non-trivial to implement with vi or emacs).

File: generate-java.sh (hunk 2 above)
Comment (jmtuley) [inline generate-java.sh:5]: Huh, I misread the go generator. Thanks for catching my error – and for making them similar.

**Comment by jmtuley** [comment]
@dsyer so it would seem that the issue with generating the Java tooling is that we import bits from gogoprotobuf, so you have to download that anyway. Am I reading that correctly?

If that's the case, then maybe we'll spend some time after Summit figuring out how to only require that library when generating Go code; it doesn't seem to me to be good policy to force Java devs through lots of hoops that have no effect on them.

**Comment by dsyer** [comment]
That would be good (not requiring `gogoprotobuf`) if it's an option, but I don't really understand what it does. I have old versions of .java files generated a while back and they still work with no reference to that stuff, so there might be a way to do that and for the generated code to still work. But then again, if they are needed for Go, why would they not be needed for Java users (it looks like it's probably just another library for Java users, but I haven't tried to make it work)?

**Comment by jmtuley** [comment]
Gogoprotobuf is the tool we use to generate Go code. The options/extensions
related to it are solely for its own use. Java users shouldn't be affected.

On Tuesday, May 12, 2015, Dave Syer notifications@github.com wrote:

> That would be good (not requiring gogoprotobuf) if it's an option, but I
> don't really understand what it does. I have old versions of .java files
> generated a while back and they still work with no reference to that stuff,
> so there might be a way to do that and for the generated code to still
> work. But then again, if they are needed for Go, why would they not be
> needed for Java users (it looks like it's probably just another library for
> Java users, but I haven't tried to make it work)?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/cloudfoundry/dropsonde-protocol/pull/2#issuecomment-101176828
> .

## 

– John Tuley

**Comment by roxtar** [comment]
@dsyer We've removed the dependency of the Java generator on the Go imports. We'd like you to apply the following diff and commit it to the PR; that way, when we merge it in, we'll be sure that Java users will benefit from your changes and not need to install Go and gogo/protobuf first.

`````` diff
diff --git a/README.md b/README.md
index f70e0b5..3eae60e 100644
--- a/README.md
+++ b/README.md
@@ -31,10 +31,7 @@ Please see the following for detailed descriptions of each type:
    ```

 ### Java
-
-1. Build the go code first (see above) so that all the imports are available
-
-2. Generate the Java code (optionally providing a target path as a directory)
+1. Generate the Java code (optionally providing a target path as a directory)
    ```
    ./generate-java.sh [TARGET_PATH]
    ```
diff --git a/generate-java.sh b/generate-java.sh
index 6afdede..a858395 100755
--- a/generate-java.sh
+++ b/generate-java.sh
@@ -19,11 +19,11 @@ fi

 pushd events
 mkdir -p $TARGET
-protoc -I=. --java_out=$TARGET --proto_path=$GOPATH/src:$GOPATH/src/github.com/gogo/protobuf/protobuf:. *.proto
+protoc -I=. --java_out=$TARGET *.proto
 popd

 pushd control
 mkdir -p $TARGET
-protoc -I=. --java_out=$TARGET --proto_path=$GOPATH/src:$GOPATH/src/github.com/gogo/protobuf/protobuf:. *.proto
+protoc -I=. --java_out=$TARGET *.proto
 popd
``````

**Comment by dsabeti** [comment]
We have removed the control messages from the dropsonde-protocol. We feel that the README changes in this PR are useful. If you modify the PR to only include those changes, we can merge this in.

**Comment by roxtar** [comment]
We have merged this PR with only the changes in the README.md and the generate-java.sh script.
